spec/ddtrace/tracer_spec.rb

```diff
@@ -0,0 +1,46 @@
+require 'spec_helper'
+
+require 'ddtrace'
+
+RSpec.describe Datadog::Tracer do
+  subject(:tracer) { described_class.new(writer: FauxWriter.new) }
+
+  describe '#trace' do
+    let(:name) { 'span.name' }
+
+    context 'given a block' do
+      subject(:trace) { tracer.trace(name, &block) }
+      let(:block) { proc { result } }
+      let(:result) { double('result') }
+
+      context 'when starting a span' do
+        it do
+          expect { |b| tracer.trace(name, &b) }.to yield_with_args(
+            a_kind_of(Datadog::Span)
+          )
+        end
+
+        it { expect(trace).to eq(result) }
+      end
+
+      context 'when starting a span fails' do
+        before(:each) do
+          allow(tracer).to receive(:start_span) do
+            raise error
+          end
+        end
+
+        let(:error) { error_class.new }
+        let(:error_class) { Class.new(StandardError) }
+
+        it 'still yields to the block and does not raise an error' do
+          expect do
+            expect do |b|
+              tracer.trace(name, &b)
+            end.to yield_with_args(nil)
+          end.to_not raise_error
```

Comment:
isn't  `to_not raise_error` a default assertion if we do not expect one to be thrown ?

Reply:
It's not really a default assertion, it just fails the test if one is thrown. What I like about doing this here is that we're really explicit in saying "this thing you'd usually expect to raise an error, doesn't actually raise an error".